WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/builders/ThreatItemBuilder.kt

```diff
@@ -0,0 +1,93 @@
+            subHeader = buildThreatItemSubHeader(threatModel),
+            onClick = { onThreatItemClicked(threatModel) }
+        )
+
+    private fun buildThreatItemHeader(threatModel: ThreatModel) = when (threatModel) {
+        is CoreFileModificationThreatModel -> UiStringResWithParams(
+            R.string.threat_item_header_infected_core_file,
+            listOf(UiStringText(getDisplayFileName(threatModel.fileName)))
+        )
+
+        is DatabaseThreatModel -> UiStringResWithParams(
+            R.string.threat_item_header_database_threat,
+            listOf(UiStringText("${threatModel.rows?.size ?: 0}"))
+        )
+
+        is FileThreatModel -> UiStringResWithParams(
+            R.string.threat_item_header_file_malicious_code_pattern,
+            listOf(UiStringText(getDisplayFileName(threatModel.fileName)))
+        )
+
+        is VulnerableExtensionThreatModel -> {
+            val slug = threatModel.extension.slug ?: ""
+            val version = threatModel.extension.version ?: ""
+            when (threatModel.extension.type) {
+                ExtensionType.PLUGIN -> UiStringResWithParams(
+                    R.string.threat_item_header_vulnerable_plugin,
+                    listOf(UiStringText(slug), UiStringText(version))
+                )
+                ExtensionType.THEME -> UiStringResWithParams(
+                    R.string.threat_item_header_vulnerable_theme,
+                    listOf(UiStringText(slug), UiStringText(version))
+                )
+                ExtensionType.UNKNOWN -> throw IllegalArgumentException(
+                    "$UNKNOWN_VULNERABLE_EXTENSION_TYPE in ${this::class.java.simpleName}"
+                )
+            }
+        }
+
+        is GenericThreatModel -> UiStringRes(R.string.threat_item_header_threat_found)
+    }
+
+    private fun buildThreatItemSubHeader(threatModel: ThreatModel) = when (threatModel) {
+        is CoreFileModificationThreatModel -> UiStringRes(R.string.threat_item_sub_header_core_file)
+
+        is DatabaseThreatModel -> UiStringText("")
+
+        is FileThreatModel -> UiStringResWithParams(
+            R.string.threat_item_sub_header_file_signature,
+            listOf(UiStringText(threatModel.baseThreatModel.signature))
+        )
+
+        is VulnerableExtensionThreatModel -> {
+            when (threatModel.extension.type) {
+                ExtensionType.PLUGIN -> UiStringRes(R.string.threat_item_sub_header_vulnerable_plugin)
+                ExtensionType.THEME -> UiStringRes(R.string.threat_item_sub_header_vulnerable_theme)
+                ExtensionType.UNKNOWN -> throw IllegalArgumentException(
+                    "$UNKNOWN_VULNERABLE_EXTENSION_TYPE in ${this::class.java.simpleName}"
+                )
+            }
+        }
+
+        is GenericThreatModel -> UiStringRes(R.string.threat_item_sub_header_misc_vulnerability)
+    }
+
+    private fun getDisplayFileName(fileName: String?) = fileName?.replace(".*/".toRegex(), "") ?: ""
```

Comment:
💡Idea: We might want to consider adding a comment with examples. It's a bit tricky to review, since I don't know what the code is supposed to do.

Reply:
Looking at the code again, it's supposed to remove the whole path except of the file name, I can see that now.

I'm wondering if we need to escape "/", wdyt?

Reply:
> I'm wondering if we need to escape "/", wdyt?

I tried by adding backslash before the forward slash. IDE is showing `illegal escape `\/`` warning.
<img width="162" alt="Screenshot 2020-12-23 at 2 22 20 PM" src="https://user-images.githubusercontent.com/1405144/102978403-73e0fe00-452a-11eb-9633-7c00210aeb3b.png">
It seems to work well in the `ThreatItemBuilderTest` tests and also displays the file name correctly in the app so looks like we don't need to escape it.